packages/credential-provider-sso/src/resolveSSOCredentials.ts

```diff
@@ -9,28 +10,49 @@ import { FromSSOInit, SsoCredentialsParameters } from "./fromSSO";
  * The time window (15 mins) that SDK will treat the SSO token expires in before the defined expiration date in token.
  * This is needed because server side may have invalidated the token before the defined expiration date.
  *
- * @internal
+ * @private
  */
 const EXPIRE_WINDOW_MS = 15 * 60 * 1000;
 
 const SHOULD_FAIL_CREDENTIAL_CHAIN = false;
 
+/**
+ * @private
+ */
 export const resolveSSOCredentials = async ({
   ssoStartUrl,
+  ssoSession,
   ssoAccountId,
   ssoRegion,
   ssoRoleName,
   ssoClient,
+  profile,
 }: FromSSOInit & SsoCredentialsParameters): Promise<Credentials> => {
   let token: SSOToken;
   const refreshMessage = `To refresh this SSO session run aws sso login with the corresponding profile.`;
-  try {
-    token = await getSSOTokenFromFile(ssoStartUrl);
-  } catch (e) {
-    throw new CredentialsProviderError(
-      `The SSO session associated with this profile is invalid. ${refreshMessage}`,
-      SHOULD_FAIL_CREDENTIAL_CHAIN
-    );
+
+  if (ssoSession) {
+    try {
+      const _token = await getSsoTokenProvider({ profile })();
+      token = {
+        accessToken: _token.token,
+        expiresAt: new Date(_token.expiration!).toISOString(),
+      };
+    } catch (e) {
+      throw new CredentialsProviderError(
+        `The SSO session ${ssoSession} for this profile is invalid. ${refreshMessage}\n` + String(e),
+        SHOULD_FAIL_CREDENTIAL_CHAIN
+      );
```

Comment:
Rethrow error message from TokenProvider here to remove redundant data?

For example, the TokenProviderError shares the REFRESH_MESSAGE for relevant errors https://github.com/aws/aws-sdk-js-v3/blob/2e13b67d85d12da2b1b12b1b83c3b80144336e22/packages/token-providers/src/fromSso.ts#L73-L77

```suggestion
      throw new CredentialsProviderError(e.message, SHOULD_FAIL_CREDENTIAL_CHAIN);
```